File: Server/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Server.Models
{
    public class Payment
    {
        [Key] 
        public int PaymentId { get; set; }
        
        [Required]
        public int LeaseId { get; set; }
        
        [Required]
        [Column(TypeName = "decimal(10, 2)")]
        public decimal Amount { get; set; }
        
        [Required]
        public DateTime PaymentDate { get; set; }
        
        [Required]
        [StringLength(20)]
        public string PaymentType { get; set; } = string.Empty; // Rent, LateFee, Deposit, Other
        
        [Required]
        [StringLength(20)]
        public string PaymentMethod { get; set; } = string.Empty; // CreditCard, BankTransfer, Cash, Check
        
        [StringLength(100)]
        public string? TransactionId { get; set; }
        
        public string? Notes { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [ForeignKey("LeaseId")]
        public Lease? Lease { get; set; }
    }
}

File: ClientApp/src/app/models/payment.ts
export interface Payment {
  id?: number;
  paymentId?: number;
  leaseId?: number;
  tenantId?: number;
  propertyId?: number;
  unitId?: number;
  amount: number;
  date?: string | Date;
  paymentDate?: string | Date;
  type?: string; // 'Rent' | 'Late Fee' | 'Deposit' | 'Other'
  paymentType?: string; // 'Rent' | 'LateFee' | 'Deposit' | 'Other'
  paymentMethod?: string; // 'CreditCard' | 'BankTransfer' | 'Cash' | 'Check'
  transactionId?: string;
  notes?: string;
  status?: string; // 'Paid' | 'Pending' | 'Failed'
  createdAt?: string | Date;
  lease?: any;
  tenantName?: string;
  unitNumber?: string;
}

export class PaymentBody {
  amount?: number;
  currency?: string;
  
   constructor(amount: number, currency: string) {
    this.amount = amount;
    this.currency = currency;
  }
}